File: RetrofitExample/app/src/main/java/com/gorkymunoz/retrofit_example/network/models/character/Character.java
package com.gorkymunoz.retrofit_example.network.models.character;

import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by Gorky Muñoz on 21/1/2021.
 * gorkymunoz@hotmail.com
 */
public class Character {

    @SerializedName("id")
    private Integer id;
    @SerializedName("name")
    private String name;
    @SerializedName("status")
    private String status;
    @SerializedName("species")
    private String species;
    @SerializedName("type")
    private String type;
    @SerializedName("gender")
    private String gender;
    @SerializedName("origin")
    private ExtraInfo originInfo;
    @SerializedName("location")
    private ExtraInfo locationInfo;
    @SerializedName("image")
    private String image;
    @SerializedName("episode")
    private List<String> episode;
    @SerializedName("url")
    private String url;
    @SerializedName("created")
    private String created;

}

File: RetrofitExample/app-kotlin/src/main/java/com/gorkymunoz/app_kotlin/network/model/character/Character.kt
package com.gorkymunoz.app_kotlin.network.model.character

import com.gorkymunoz.app_kotlin.network.model.ExtraInfo
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass


/**
 * Created by Gorky Muñoz on 26/7/2021.
 *
 *
 */
@JsonClass(generateAdapter = true)
data class Character(
    val id: Int,
    val name: String,
    val status: String,
    val species: String,
    val type: String,
    val gender: String,
    val origin: ExtraInfo,
    val location: ExtraInfo,
    val image: String,
    @Json(name = "episode") val listOfEpisodes: List<String>,
    val url: String,
    val created: String
)
